Oracle know rep sta2 july2023

SalenaAshton opened the pull request.

skdebray committed:
initial commit

skdebray committed:
minor edits

skdebray committed:
some python code for will creation

skdebray committed:
renamed file

skdebray committed:
added various refinements

skdebray committed:
added various refinements

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
moved some python code to wills.py

skdebray committed:
typo fix

skdebray committed:
added some design considerations

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
minor restructuring

skdebray committed:
python code to create a will, and updated design ideas

skdebray committed:
some basic usage information

skdebray committed:
minor prettification

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
minor edits

skdebray committed:
renamed the file in01.json as will01.json

skdebray committed:
minor edits

skdebray committed:
added stubs for validation checks

skdebray committed:
minor edits

skdebray committed:
code update to remove redundant validity checks on witnesses

Commit:
finish the filler functions

Commit:
typo fixed

Commit:
typo fixed

Commit:
fix pickle extension, add filename to export's print output

skdebray committed:
minor tweaks

skdebray committed:
minor tweaks

skdebray committed:
minor tweaks to JSON

skdebray committed:
changed JSON for witnesses to be a list

skdebray committed:
minor bug fix

skdebray committed:
made the invocation interface to create_wills() more modular

skdebray committed:
minor refactoring

skdebray committed:
added two new wills, will02.json and will03.json, with slightly more complex constructs

skdebray committed:
stashing JSON files for wills into a single directory

Commit:
add devolve_will.py, which loads the will, validates its hash, validates each directive, and executes the validated directives.

Commit:
add devolve_will.py, which loads the will, validates its hash, validates each directive, and executes the validated directives.

cl4yton committed:
avoid committing files/directories generated by interpreter

cl4yton committed:
first commit

skdebray committed:
added a "date" field to each JSON file

skdebray committed:
added some validation code + sample oracle db

Commit:
add formatting

Commit:
remove extra imports

Commit:
fix line length to 79

skdebray committed:
line length adjustments for PEP-8 style guidelines

SalenaAshton committed:
Added constraints, quantifiers and other representations. Suggested date validation as a cheap way to validate wills at this stage. Some keys have values called placeholders. In progress.

SalenaAshton committed:
Update README.md

SalenaAshton committed:
minor revision in change log

SalenaAshton committed:
Corrected json file so it will parse correctly. Comments at top of file will eventually need to be removed.

SalenaAshton committed:
Added death date of testator, bequeathed_items keys to persons, and two additional items to be bequeathed.

SalenaAshton committed:
update

cl4yton committed:
refactor

cl4yton committed:
basic simple model

cl4yton committed:
making naming consistent

cl4yton committed:
reintroducing ORACLES

cl4yton committed:
attempting to pull

SalenaAshton committed:
Arwood tree and family json refactored.

SalenaAshton committed:
Arwood tree and family json refactored.

SalenaAshton committed:
Refactored will and family tree jsons for Leo Arwood.

cl4yton committed:
first commit

alicekwak committed:
Upload extraction files created from Arwood's will

skdebray commented on ORACLES/db01_sta.json:
```diff
@@ -0,0 +1,145 @@
+/* SALENA 2 July 2023         Comments about changes that I've made
```
JSON doesn't allow comments, so this is not legal JSON.  I think it would be desirable for a file that has a ".json" extension to be legal JSON.  Would it be possible to either remove comments or else move JSON-plus-comments files to a `md` format?

SalenaAshton committed:
Update db01_sta.json

Removed comments from json file.

SalenaAshton commented:
Hi Saumya!

I just removed the comments and committed the changes.

Thanks!
Salena

On Wed, Jul 19, 2023 at 4:55 PM Saumya Debray ***@***.***>
wrote:

> *External Email*
>
> ***@***.**** commented on this pull request.
> ------------------------------
>
> In ORACLES/db01_sta.json
> <https://github.com/ml4ai/dass-wills/pull/3#discussion_r1268791495>:
>
> > @@ -0,0 +1,145 @@
> +/* SALENA 2 July 2023         Comments about changes that I've made
>
> JSON doesn't allow comments, so this is not legal JSON. I think it would
> be desirable for a file that has a ".json" extension to be legal JSON.
> Would it be possible to either remove comments or else move
> JSON-plus-comments files to a md format?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ml4ai/dass-wills/pull/3#pullrequestreview-1538210076>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANUDNJR2A2LJJWEEFSK2XVDXRBXXRANCNFSM6AAAAAA2AXZAIQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

skdebray commented on ORACLES/db01_sta.json:
```diff
@@ -0,0 +1,124 @@
+            }
+    }
+}
+}
```
These curly braces are unbalanced, so this file is not syntactically correct.  This needs to be fixed.  (**Note:** I didn't just add a bunch of right-curly-braces at the end because I did not know how this would affect which values are associated with which fields.  That needs to be checked.)

REQUEST: Please check any updated JSON files to make sure that they are syntactically correct JSON files.

SalenaAshton committed:
Balanced out curly braces at end of file and double-checked the fields.

SalenaAshton committed:
Delete db01_sta.json

I need think about the representation that the lists suggest, so it is better not to push this file on main.

SalenaAshton commented:
Sorry about that!

Yeah, it looks like there had been one extra curly brace at the end of a
field. Instead of fixing it, I've deleted this whole file so it won't be
pushed onto main. Clay and I talked about removing this file so we can
discuss what I'd like to represent in a list versus a dictionary.

Thanks!
Salena

On Wed, Jul 19, 2023 at 5:15 PM Saumya Debray ***@***.***>
wrote:

> *External Email*
>
> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In ORACLES/db01_sta.json
> <https://github.com/ml4ai/dass-wills/pull/3#discussion_r1268801840>:
>
> > +                "asset_id" : "personal_vac_01",
> +                "asset_name" : "vacuum",
> +                "asset_validation" : "Place-holder",
> +                "testator_status" : "dead",
> +                "beneficiary_isAvail" : true,
> +                      "if" : true,
> +                      "then" : {
> +                        "beneficiary" : {
> +                            "name" : "Mary Hoover",
> +                            "id" : "0x5678",
> +                            "info" : "place holder" }
> +                      }
> +            }
> +    }
> +}
> +}
>
> These curly braces are unbalanced, so this file is not syntactically
> correct. This needs to be fixed. (*Note:* I didn't just add a bunch of
> right-curly-braces at the end because I did not know how this would affect
> which values are associated with which fields. That needs to be checked.)
>
> REQUEST: Please check any updated JSON files to make sure that they are
> syntactically correct JSON files.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ml4ai/dass-wills/pull/3#pullrequestreview-1538227279>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANUDNJWTTI3KN2OXK5MWSHDXRB2AFANCNFSM6AAAAAA2AXZAIQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

skdebray commented:
@SalenaAshton @pawnsac @cl4yton I was about to merge this PR when I realized that it wants to merge `oracle_knowRep_sta2_July2023` into `main`.  But I don't think the Python code for executing and devolving wills has not been updated to work with the new JSON format, and I'd like everything in `main` to be consistent and working as far as possible, so I'd like to have this problem fixed before I merge this PR into `main`.  

I will go ahead and create a new PR to merge `oracle_knowRep_sta2_July2023` into `design-attempt-01`, after which:

- @Muaz: I'll create an issue for fixing the Python code, which I'll assign to you.  
- @SalenaAshton: If the JSON format has now been nailed down, maybe this should be documented somewhere and `README.md` updated to point to it?
- **All of us**: We're at (or very close to) the point where we can create small unit tests for our code.  The wills we've created so far have all been legal.  But maybe we should also think about creating a set of not-legal wills, e.g., where the testator and/or witnesses don't satisfy the legal requirements, so that we can test the validation conditions.
